src/org/omegat/core/machinetranslators/MTHUBTranslate.java

```diff
@@ -0,0 +1,313 @@
+        return OStrings.getString("MT_ENGINE_MTHUB");
+    }
+
+    @Override
+    public boolean isConfigurable() {
+        return true;
+    }
+
+    @Override
+    public void showConfigurationUI(Window parent) {
+        MTConfigDialog dialog = new MTConfigDialog(parent, getName()) {
+            @Override
+            protected void onConfirm() {
+                String key = panel.valueField1.getText().trim();
+                boolean temporary = panel.temporaryCheckBox.isSelected();
+                setCredential(PROPERTY_API_KEY, key, temporary);
+            }
+        };
+        dialog.panel.valueLabel1.setText(OStrings.getString("MT_ENGINE_MTHUB_API_KEY_LABEL"));
+        dialog.panel.valueField1.setText(getCredential(PROPERTY_API_KEY));
+        dialog.panel.valueLabel2.setVisible(false);
+        dialog.panel.valueField2.setVisible(false);
+        dialog.panel.temporaryCheckBox.setSelected(isCredentialStoredTemporarily(PROPERTY_API_KEY));
+        dialog.show();
+    }
+
+    @Override
+    protected String translate(Language sLang, Language tLang, String text) throws Exception {
+        if (this.availableLanguageCodes.isEmpty()) {
+         this.setAvailableLanguageCodes();
+        }
+
+        String mthubKey = getCredential(PROPERTY_API_KEY);
+        if (mthubKey == null || mthubKey.isEmpty()) {
+            return OStrings.getString("MT_ENGINE_MTHUB_KEY_NOTFOUND");
+        }
+
+        String prev = getFromCache(sLang, tLang, text);
+        if (prev != null) {
+            return prev;
+        }
+        String charset = "UTF-8";
+
+        try {
+            URLConnection connection = new URL(GT_URL).openConnection();
+            connection.setDoOutput(true);
+            connection.setRequestProperty("Accept-Charset", charset);
+            connection.setRequestProperty("Content-Type", "application/json ;charset=" + charset);
+
+            OutputStream output = connection.getOutputStream();
+
+            ObjectMapper om = new ObjectMapper();
+
+            ArrayList<String> segments = new ArrayList<String>();
+            segments.add(text.substring(0, Math.min(text.length(), LIMIT_CHARACTER - 1)));
+
+            MTHUBTranslateRequest mttr = new MTHUBTranslateRequest(
+                    mthubKey, getNormaliseCode(sLang), getNormaliseCode(tLang), segments);
+
+            String json = om.writeValueAsString(mttr);
+            output.write(json.getBytes(charset));
+
+            Map<String, Object> map = om.readValue(connection.getInputStream(),
+                                      new TypeReference<Map<String, Object>>() { });
+            String translation = getJsonResults(map);
+
+            putToCache(sLang, tLang, text, translation);
+            return translation;
+        } catch (MalformedURLException e) {
+            return StringUtil.format(OStrings.getString("MTHUB_ERROR"), e.getMessage());
+        } catch (Exception e) {
+            if (e.getMessage().contains("401")) {
```

Comment:
Actually the `connection` object will always be an instance of `HTTPURLConnection` and you can get the response code like so:

`((HttpURLConnection) connection).getResponseCode()`

You should be checking this up above rather than parsing the error message.